lib/fluent/plugin/buf_file.rb

```diff
@@ -56,6 +51,14 @@ def initialize
       def configure(conf)
         super
 
+        unless @path
```

Comment:
Anyway, `plugin_id` can't help that situation. `plugin_id` is (should be) unique in a worker process, not server global.